packages/connector-wechat-native/src/index.ts

```diff
@@ -92,25 +90,27 @@ export default class WeChatNativeConnector implements SocialConnector {
         })
         .json<UserInfoResponse>();
 
-      if (!openid || errcode || errmsg) {
-        // 'openid' is defined as a required input argument in WeChat API doc, but it does not necessarily to
-        // be the return value from getAccessToken per testing.
-        // In another word, 'openid' is required but the response of getUserInfo is consistent as long as
-        // access_token is valid.
-        // We are expecting to get 41009 'missing openid' response according to the developers doc, but the
-        // fact is that we still got 40001 'invalid credentials' response.
-        if (errcode === 40_001) {
-          throw new ConnectorError(ConnectorErrorCodes.SocialAccessTokenInvalid);
-        }
+      // 'openid' is defined as a required input argument in WeChat API doc, but it does not necessarily to
+      // be the return value from getAccessToken per testing.
+      // In another word, 'openid' is required but the response of getUserInfo is consistent as long as
+      // access_token is valid.
+      // We are expecting to get 41009 'missing openid' response according to the developers doc, but the
+      // fact is that we still got 40001 'invalid credentials' response.
 
-        throw new Error(errmsg);
-      }
+      // Response properties of user info can be separated into two groups: (1) {unionid, headimgurl, nickname}, (2) {errcode, errmsg}.
+      // These two groups are mutually exclusive: if group (1) is not empty, group (2) should be empty and vice versa.
+      // 'errmsg' and 'errcode' turn to be non-empty values or empty values at the same time. Hence, if 'errmsg' is non-empty then 'errcode' should be non-empty.
+
+      assert(errcode !== 40_001, new ConnectorError(ConnectorErrorCodes.SocialAccessTokenInvalid));
+      assert(!errcode, new Error(errmsg));
+      assert(openid, new Error(errmsg));
```

Comment:
Looks confusing here, maybe guard it when we first get this `openid` in line: 84.

Reply:
Good call, fixed.